View3D/Services/SimplygonHelpers.cs

```diff
@@ -0,0 +1,190 @@
+﻿using Simplygon;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using View3D.Rendering.Geometry;
+using SharpDX.Direct3D9;
+using Microsoft.Xna.Framework;
+
+namespace View3D.Services
+{
+    /// <summary>
+    /// Collections of methods to interact with simplygon, do the dirt work
+    /// </summary>
+    public class SimplygonHelpers
+    {
+        public static void InitSGGeometryDataObject(ISimplygon sg, out spGeometryData geometryDataSG, MeshObject originalMesh)
+        {
+            // -- create SG GeomtryData object and allocate space for mesh
+            geometryDataSG = sg.CreateGeometryData();            
+            geometryDataSG.SetTriangleCount(((uint)originalMesh.GetIndexCount()) / 3);
+            geometryDataSG.SetVertexCount((uint)originalMesh.GetVertexList().Count);
+
+            geometryDataSG.AddBoneWeights(4); 
+
+            geometryDataSG.AddNormals();
+            geometryDataSG.AddTangents(0);
+
+            geometryDataSG.AddTexCoords(0); // add two texcoord channels, if format is "default" (static props)
+            geometryDataSG.AddTexCoords(1); // TODO: AE should support 2 texture channel as some "default"/static models use it
+
+        }
+        public static void FillSGVertices(ref spGeometryData geometryDataSG, MeshObject originalMesh)
```

Comment:
Consider not naming the functions SG. It add no real value and makes it harder to read